client/src/components/molecules/RequestTypeDropdown.tsx

```diff
@@ -0,0 +1,161 @@
+import { gql, useMutation } from '@apollo/client';
+import React, { FunctionComponent, useState } from 'react';
+import CommonModal from '../organisms/Modal';
+import Dropdown from '../atoms/Dropdown';
+import Request from '../../data/types/request';
+import RequestGroup from '../../data/types/requestGroup';
+import RequestsTable from './RequestsTable';
+import RequestType from '../../data/types/requestType';
+
+interface Props {
+    key?: string;
+    requestType?: RequestType;
+    requestGroup?: RequestGroup;
+    requests?: Request[];
+}
+
+const RequestTypeDropdown: FunctionComponent<Props> = (props: Props) => {
+    const mutation = gql`
+    mutation updateRequestType($requestType: RequestTypeInput){
+        updateRequestType(requestType: $requestType){
+          id
+          success
+          message
+        }
+      }
+    `;
+
+    const softDelete = gql`
+    mutation deleteARequestType($id: ID){
+        softDeleteRequestType(id: $id){
+          id
+          success
+          message
+        }
+      }`;
+
+    const [requestType, setRequestType] = useState(props.requestType);
+    const [backupRequestType, setBackupRequestType] = useState(requestType);
+    const [editModalShow, setEditModalShow] = useState(false);
+    const [deleteModalShow, setDeleteModalShow] = useState(false);
+    const [isTooLong, setIsTooLong] = useState(false);
+    const [mutateRequestType, { error }] = useMutation(mutation);
+    const [mutateDeleteRequestType] = useMutation(softDelete);
+
+    const getTotalQuantity = () => {
+        let totalQuantity = 0;
+        requestType?.requests!.forEach(request => {
+            totalQuantity += request.quantity!
+        });
+        return totalQuantity;
+    }
+
+    const editModalTitle = "Edit Type";
+    const deleteModalTitle = "Delete Type";
+    const deleteModalSubtitle = "Are you sure you want to delete "  + requestType!.name + " as a type in the group " + props.requestGroup!.name + "? This will delete all "+ getTotalQuantity() + " requests within this type and cannot be undone."
+    
+    const tooLongMessage = "Type name cannot exceed 40 characters!";
+
+    const handleEditModalClose = () => {
+        setRequestType(backupRequestType);
```

Comment:
Can't tell what backupRequestType is being used for. Maybe instead, just don't modify the underlying requestType until the user confirms their action, just modify the UI.

Reply:
I used backupRequestType to store the state of requestType before its being edited. So if user is editing requestType and decides they want to exit the modal and not save changes. Then the UI will not show the messed up requestType but instead the backupRequestType. It won't save but I think its nice to have it in the UI

Reply:
@Anand-Issac Don't see anywhere in the code that uses this